### ISSUE TITLE
Fastlane pilot - upload to testflight error

### ISSUE THREAD

**Issue by wlmpoc** [fastlane/fastlane#12411]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Fastlane crashes after uploading an ipa file to testflight.

From my observation, it looks like once upload is successful, there is a slight delay for the processing to start and the ipa file (Processing) status does not appear immediately in iTunesConnect. Within this time the fastlane build_watcher.rb crashes with error **Could not find a build for app: **** on platform: ios**.

The ipa file shows up in iTunesConnect after a few minutes with (Processing) status and finally processed successfully.

The testgroup is not updated as fastlane crashed instead of waiting in loop for processing to complete. So we had to update the test group manually.

### Some additional points:-

- The size of the ipa file is close to 160Mb.
- In fastlane code, **_we first expire the current build and then use pilot to upload_** the new ipa file
 
### Complete output when running fastlane, including the stack trace and command used:-

<details>
  <pre>[INFO [2018-04-17 10:16:53.24]: [32m-----------------------------------------------------[0m
INFO [2018-04-17 10:16:53.24]: [32mSuccessfully uploaded package to iTunes Connect. It might take a few minutes until it's visible online.[0m
INFO [2018-04-17 10:16:53.24]: [32m-----------------------------------------------------[0m
INFO [2018-04-17 10:16:53.25]: [32mSuccessfully uploaded the new binary to iTunes Connect[0m
INFO [2018-04-17 10:16:53.25]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
WARN [2018-04-17 10:16:56.31]: [33m[33mLane Context:[0m
INFO [2018-04-17 10:16:56.31]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios betaupload", :GET_IPA_INFO_PLIST_VALUE_CUSTOM_VALUE=>"3.xx", :LATEST_BUILD_NUMBER=>41, :LATEST_TESTFLIGHT_BUILD_NUMBER=>41}
ERROR [2018-04-17 10:16:56.31]: [31mCould not find a build for app: **** on platform: ios[0m
INFO [2018-04-17 10:16:56.31]: [32mSuccessfully generated documentation at path 

+------+------------------------+-------------+
|              [32mfastlane summary[0m               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | Verifying fastlane     | 0           |
|      | version                |             |
| 2    | default_platform       | 0           |
| 3    | get_ipa_info_plist_va  | 0           |
|      | lue                    |             |
| 4    | get_ipa_info_plist_va  | 0           |
|      | lue                    |             |
| 5    | latest_testflight_bui  | 5           |
|      | ld_number              |             |
| ðŸ’¥   | [31mpilot[0m                  | 351         |
+------+------------------------+-------------+

ERROR [2018-04-17 10:16:56.33]: [31mfastlane finished with errors[0m

/Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane_core/lib/fastlane_core/ui/interface.rb:121:in `crash!': [31m[!] Could not find a build for app: **** on platform: ios[0m (FastlaneCore::Interface::FastlaneCrash)
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane_core/lib/fastlane_core/build_watcher.rb:13:in `wait_for_build_processing_to_be_complete'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/pilot/lib/pilot/build_manager.rb:55:in `upload'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/actions/upload_to_testflight.rb:16:in `run'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:231:in `chdir'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
	from Fastfile:106:in `block (2 levels) in parsing_binding'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/lane_manager.rb:59:in `cruise_lane'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:75:in `run!'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/commands_generator.rb:333:in `run'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/fastlane/lib/fastlane/cli_tools_distributor.rb:107:in `take_off'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/lib/ruby/gems/2.2.0/gems/fastlane-2.92.0/bin/fastlane:20:in `<top (required)>'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/bin/fastlane:22:in `load'
	from /Users/xxxxxxxxxxxxxxxx/.fastlane/bin/bundle/bin/fastlane:22:in `<main>']</pre>
</details>

### Environment

<details>
  <pre>[mymac:fastlane xxxxxxxxxxxxxxxxxx$ /Users/xxxxxxxxxxxxxxxxxx/.fastlane/bin/fastlane env

fastlane could not check for updates error: execution expired
[06:17:44]: Generating fastlane environment output, this might take a few seconds...
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 10.11.6                                          |
| Ruby                        | 2.2.4                                            |
| Bundler?                    | false                                            |
| Git                         | git version 2.10.1 (Apple Git-78)                |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane              |
| Host                        | Mac OS X 10.11.6 (15G17023)                      |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                       |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                       |
| Is contained                | true                                             |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode8.2.1.app/Contents/Developer/ |
| Xcode Version               | 8.2.1                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

app_identifier "xxxxxxxxxxxxxxxxxx" # The bundle identifier of your app
apple_id "xxxxxxxxxxxxxxxxxx" # Your Apple email address

#team id is not required.
#team_id "xxxxxxxxxxxxxxxxxx" # Developer Portal Team ID

#you can even provide different app identifiers, Apple IDs and team names per lane:
#More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

</details>

### fastlane gems

| Gem      | Version | Update-Status  |
| -------- | ------- | -------------- |
| fastlane | 2.92.0  | 💥 Check failed |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| CFPropertyList            | 2.3.5        |
| addressable               | 2.5.1        |
| atomos                    | 0.1.2        |
| babosa                    | 1.0.2        |
| bigdecimal                | 1.2.6        |
| bundler                   | 1.14.6       |
| claide                    | 1.0.2        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.0        |
| emoji_regex               | 0.1.1        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.1.2        |
| google-api-client         | 0.13.6       |
| googleauth                | 0.5.1        |
| highline                  | 1.7.8        |
| http-cookie               | 1.0.3        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| io-console                | 0.4.3        |
| json                      | 1.8.1        |
| jwt                       | 1.5.6        |
| libxml-ruby               | 3.0.0        |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| memoist                   | 0.15.0       |
| mime-types                | 3.1          |
| mime-types-data           | 3.2016.0521  |
| mini_magick               | 4.5.1        |
| minitest                  | 5.4.3        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| multipart-post            | 2.0.0        |
| nanaimo                   | 0.2.4        |
| naturally                 | 2.1.0        |
| nokogiri                  | 1.5.6        |
| os                        | 0.9.6        |
| plist                     | 3.2.0        |
| power_assert              | 0.2.2        |
| psych                     | 2.0.8        |
| public_suffix             | 2.0.5        |
| rake                      | 10.4.2       |
| rdoc                      | 4.2.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| rouge                     | 1.11.1       |
| rubygems-update           | 2.5.2        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| signet                    | 0.7.3        |
| simctl                    | 1.6.3        |
| slack-notifier            | 2.3.2        |
| sqlite3                   | 1.3.13       |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.7.3        |
| test-unit                 | 3.0.8        |
| tty-cursor                | 0.5.0        |
| tty-screen                | 0.6.3        |
| tty-spinner               | 0.8.0        |
| uber                      | 0.0.15       |
| unf                       | 0.1.4        |
| unf_ext                   | 0.0.7.4      |
| unicode-display_width     | 1.1.3        |
| word_wrap                 | 1.0.0        |
| xcodeproj                 | 1.5.7        |
| xcpretty                  | 0.2.6        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2018-04-24**
</details>
[06:18:44]: Take notice that this output may contain sensitive information, or simply information that you don't want to make public.
[06:18:44]: 🙄  Wow, that's a lot of markdown text... should fastlane put it into your clipboard, so you can easily paste it on GitHub? (y/n)
]</pre>
</details>

**Comment by joshdholtz** [comment]
@wlmpoc This looks like it could be an issue similar to this one - https://github.com/fastlane/fastlane/issues/12060. Does this happen to you on a consistent basis? If I make an attempt at fixing this for you, would you be able to test it out? 😊

**Comment by wlmpoc** [comment]
Good day to you joshdholtz

We ran only two deployments using fastlane till now. Before that we were doing only Manual deployments.
Both the deployments failed with the same issue. We saw that the app was there in iTunesConnect and we manually moved it to Beta Testers group
We will be running one more "Test flight" upload today. I will let you know once it is done.

**Comment by illl48** [comment]
Hi @joshdholtz , once yo fix this issue I would  be glad to test it. Thank you.

**Comment by wlmpoc** [comment]
Hi joshdholtz

I ran some tests today.  Explaining this in order below:  

1. We wanted to try what would happen if we exipred the previous build after the pilot is successful and hence we **moved** the _expire()!_ call **below** _pilot()_ instead of above pilot() in our Fastfile
2.  Tried "push to testflight"
3.  Observed that build_watcher detected the build (old one) and so did not crash. But did not enter into the wait loop either. It directly tried to submit build (old one) itself to testers and Failed with error pasted below: <br>
<details>
<pre>INFO [2018-05-03 03:33:41.80]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
WARN [2018-05-03 03:33:46.87]: [33mStarted watching build 2.20 - 44 but expected 2.20 - 46[0m
INFO [2018-05-03 03:33:47.90]: [32mBuild 2.20 - 44 is already being tested[0m
WARN [2018-05-03 03:33:47.90]: [33mUploaded app 2.20 - 46, but received build 2.20 - 44. If you want to wait for uploaded build to be finished processing, use the `wait_for_uploaded_build` option[0m
INFO [2018-05-03 03:33:52.34]: [32mSuccessfully set the changelog for build[0m
INFO [2018-05-03 03:33:52.34]: Distributing new build to testers: 2.20 - 44WARN [2018-05-03 03:33:53.29]: [33m[33mLane Context:[0m
INFO [2018-05-03 03:33:53.29]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios betaupload", :GET_IPA_INFO_PLIST_VALUE_CUSTOM_VALUE=>"2.20", :LATEST_BUILD_NUMBER=>44, :LATEST_TESTFLIGHT_BUILD_NUMBER=>44}</pre></details><br>

4. So again we added the option wait_for_uploaded_build to pilot() and tried.
5. As expected, this run did enter into wait loop. But since there was an issue with the IOS app, the build "Processing" in itunes connect failed and we thought this loop would wait forever and hence killed the fastlane process

Please let me know if you need any more inputs
